### PR TITLE
CB-8053 Enable RAZ for AWS

### PR DESCRIPTION
**Pull request by lajosrodek**
* Refactor `AccountMappingSubject`: extract user `rangerraz` as the new constant `RANGER_RAZ_USER`, rename `DATA_ACCESS_AND_RANGER_AUDIT_USERS` to `ALL_SPECIAL_USERS`.
* Add S3 to `ModelDescriptions.ClusterModelDescription.ENABLE_RANGER_RAZ`.
* Extend cloud provider restriction with AWS in `RangerCloudStorageServiceConfigProvider`, `RangerRazDatahubConfigProvider`, `RangerRazDatalakeConfigProvider` and `SdxService.validateRazEnablement`.
* Change CM version check in `RangerRazDatalakeConfigProvider` to use 7.2.2 for AWS and 7.2.1 for Azure.
* Add CM 7.2.2 version check to `RangerRazDatahubConfigProvider` (to make it in sync with `RangerRazDatalakeConfigProvider`).
* Update all related unit tests and add some new cases.

### CODE REVIEW

**Comment by lajosrodek** [comment]
Regarding the `rangerraz` refactor in `AccountMappingSubject`: follow-up to the earlier change #8183.

File: datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java [hunk 1]
```diff
@@ -440,11 +441,14 @@ private void validateRazEnablement(SdxClusterRequest sdxClusterRequest, Detailed
             if (!razEntitlementEnabled) {
                 validationBuilder.error("Provisioning Ranger Raz is not enabled for this account.");
             }
-            if (!AZURE.name().equalsIgnoreCase(environment.getCloudPlatform())) {
-                validationBuilder.error("Provisioning Ranger Raz is only valid for Microsoft Azure.");
+            if (!(AWS.name().equalsIgnoreCase(environment.getCloudPlatform()) || AZURE.name().equalsIgnoreCase(environment.getCloudPlatform()))) {
```
Comment (orngejaket) [inline datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java:444]: We reuse the ignoresCase a lot with treating checks for the cloud platform.  Lets put that in a variable so we can just do an enum comparison.

File: datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java [hunk 2]
```diff
@@ -440,11 +441,14 @@ private void validateRazEnablement(SdxClusterRequest sdxClusterRequest, Detailed
...
+                validationBuilder.error("Provisioning Ranger Raz is only valid for Amazon Web Services and Microsoft Azure.");
             }
-            if (!isRazSupported(sdxClusterRequest.getRuntime())) {
-                validationBuilder.error("Provisioning Ranger Raz is only valid for CM version >= 7.2.1 and not " + sdxClusterRequest.getRuntime());
+            if (!isRazSupported(sdxClusterRequest.getRuntime(), environment.getCloudPlatform())) {
+                boolean aws = AWS.name().equalsIgnoreCase(environment.getCloudPlatform());
+                String errorMsg = aws ? "Provisioning Ranger Raz on Amazon Web Services is only valid for CM version >= 7.2.2 and not " :
```
Comment (orngejaket) [inline datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java:449]: we could do AWS.equals(platform) ? "" : "";  where platform is defined as my previous comment.

File: datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java (hunk 1 above)
Comment (cegganesh84) [reply datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java]: Thanks, addressed.

File: datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java (hunk 2 above)
Comment (cegganesh84) [reply datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java]: Thanks, addressed.

**Comment by cegganesh84** [comment]
@doktoric we want this feature for 2.28, should I set the base branch to 2.28. I see that it is branched but not deployed to mow-dev yet.

**Comment by cegganesh84** [comment]
This change will not cleanly merge with 2.28 because there are changes to SdxService files that are not in the branch. I will send a separate PR for 2.28.

**Comment by cegganesh84** [comment]
Sent PR #8872 for 2.28 branch.

**Comment by lajosrodek** [comment]
@cegganesh84 Looks like your #8872 has been merged into 2.28, 2.29 and 2.30 (current master), so I suppose we do not actually need this PR anymore. 😃 I am happy to abandon this one unless you disagree.

**Comment by cegganesh84** [comment]
@lajosrodek Yes, we can abandon this. Thanks!

**Pull request closed without merging**